### PR TITLE
Persist column state+ scale column-zoom

### PR DESCRIPTION
**Pull request by UncleFairy**
#### What's this PR do?

Resist columnDef Problem
...

#### Sanity Check:
- [ ] Tested on IE 11.
- [ ] Tested on Chrome.
- [ ] Tested on Firefox.
- [ ] Unit tests

### CODE REVIEW

File: src/panels/ExchangeRates/index.js [hunk 1]
```diff
@@ -34,9 +36,11 @@ const ExchangeRates = ({
   setCurrency,
   setZoomValue,
   zoomValue,
+  columnState,
+  columnsUpdate,
 }) => {
-  const [gridApi, setGridApi] = useState(null)
-  const [columns, setColumnsWidth] = useState(columnDefs)
+  const [apiCallback, setApiCallback] = useState(null)
+  const [prevZoom, setPrevZoom] = useState(zoomValue)
```
Comment (newalex997) [inline src/panels/ExchangeRates/index.js]: why you need 'prevZoom'?

File: src/panels/ExchangeRates/index.js [hunk 2]
```diff
@@ -82,6 +82,27 @@ const ExchangeRates = ({
     setCurrency({ meta, payload })
   }
 
+  const onColumnChanged = ({ columnApi }) => {
```
Comment (newalex997) [inline src/panels/ExchangeRates/index.js]: move all methods to separated 'utils' file

File: src/panels/ExchangeRates/index.js [hunk 3]
```diff
@@ -94,10 +115,12 @@ const ExchangeRates = ({
         zoomValue={zoomValue}
       />
       <Grid
-        columnDefs={columns}
+        columnDefs={columnDefs}
         rowData={rowData}
-        onGridReady={setGridApi}
-        rowHeight={newSizeCalculator(zoomDefaultSizes.height, zoomValue)}
+        onGridReady={onGridReady}
+        rowHeight={newHeightCalculator(cellDefaultSizes.height, zoomValue)}
```
Comment (newalex997) [inline src/panels/ExchangeRates/index.js]: do not pass functions as props values directly. Save `newHeightCalculator(cellDefaultSizes.height, zoomValue)` in constant and pass this as prop value

File: src/panels/ExchangeRates/index.js (hunk 1 above)
Comment (UncleFairy) [reply src/panels/ExchangeRates/index.js]: I need 'prevZoom' for 'newWidthCalculator' function to calculate correct new width for column with custom width

File: src/panels/ExchangeRates/utils.js [hunk 4]
```diff
@@ -3,5 +3,15 @@ export const currencyFormatter = params => {
...
+
+export const setNewState = (newState, columnApi) => {
+  try {
+    columnApi.setColumnState(newState)
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+export const getCurrentState = columnApi => columnApi.getColumnState()
```
Comment (newalex997) [inline src/panels/ExchangeRates/utils.js]: this is useless

File: src/panels/ExchangeRates/index.js [hunk 5]
```diff
@@ -82,6 +81,12 @@ const ExchangeRates = ({
     setCurrency({ meta, payload })
   }
 
+  const onColumnChanged = ({ columnApi }) =>
+    columnsUpdate({
+      payload: { columnState: getCurrentState(columnApi) },
```
Comment (newalex997) [inline src/panels/ExchangeRates/index.js]: ` payload: columnApi.getColumnsState()`

File: src/redux/actions/index.js [hunk 6]
```diff
@@ -50,3 +51,14 @@ export const changeZoomValue = ({ payload, meta }) => ({
   payload,
   meta,
 })
+
+export const columnsUpdate = ({ payload, meta }) => ({
+  type: COLUMNS_UPDATE,
+  payload,
+  meta: {
+    ...meta,
+    debounce: {
```
Comment (newalex997) [inline src/redux/actions/index.js]: `onColumnMoved` have `finished` callback param, you should update reducer store just when `finished` param exists. And debounce will be useless

**Pull request merged**